lib/quiet_quality/version_control_systems/git.rb

```diff
@@ -39,7 +57,48 @@ def self.default_branch(remote:)
       # @return [String] The nearest common ancestor (SHA)
       #
       def comparison_base(sha:, comparison_branch:)
-        git.merge_base(comparison_branch, sha)
+        git.merge_base(comparison_branch, sha).first
+      end
+
+      private def changed_lines_for(diff)
+        GitDiffParser.parse(diff).flat_map do |parsed_diff|
+          parsed_diff.changed_line_numbers.to_set
+        end
+      end
+
+      private def lines_in_file(file_path)
+        File.open(file_path, "r").readline.count
```

Comment:
Will this leave the file open? We might need to use the block form, like

```
File.open(file_path, "r") do |file|
  return file.readline.count
end
```

Reply:
I've added support for 'entirely' changed files [here](https://github.com/nevinera/quiet_quality/pull/26), and for `ChangedFiles#merge` [here](https://github.com/nevinera/quiet_quality/pull/27).

The latter's not critical - if you'd rather keep that merging complexity inside the git implementation, it's reasonably likely nothing else will actually need it. But I think it's neat :-)

Reply:
👍🏼 will take a look. Yeah this is only needed for untracked files, as we need to know the line count. I was initially shelling out to `wc`, but that felt like cheating. I think I intended to call `readline**s**` here! Would have caught it when I got testing!